src/MSRIOGroup.cpp

```diff
@@ -518,6 +563,19 @@ namespace geopm
         if (control_name == "POWER_PACKAGE_LIMIT") {
             write_control("MSR::PKG_POWER_LIMIT:PL1_LIMIT_ENABLE", domain_type, domain_idx, 1.0);
         }
+
+        //HWP Handling
+        if (m_hwp_is_enabled) {
```

Comment:
Same general approach as POWER_PACKAGE_LIMIT writes implying the P1_LIMIT_ENABLE, but the approach is messy long term if more signals require it.  A combined control object may resolve this.